src/clib/pio_darray.c

```diff
@@ -226,8 +249,13 @@ int pio_write_darray_nc(file_desc_t *file, io_desc_t *iodesc, const int vid,
 #endif /* _NETCDF4 */
 	    case PIO_IOTYPE_NETCDF:
 	    {
+		/* Find the type size (again?) */
 		mpierr = MPI_Type_size(iodesc->basetype, &dsize);
+		
 		size_t tstart[ndims], tcount[ndims];
+
+		/* The IO master task does all the data writes, but
```

Comment:
When using the serial netcdf PIO acts like a funnel.   Data is moved from compute tasks to IO tasks then from IO tasks to IO task 0 for write.  For read data the opposite happens, data is read on IO task 0 distributed to all io tasks and then to compute tasks.   We could optimize these data paths but serial netcdf is not the primary mode for PIO.   Can you change this comment?